Super admin

OwenLobato opened the pull request:
- Fix superadmin login
- Improve organization form design
- Superadmin can update organization, admin and passphrase data

OwenLobato committed:
Add organization data to edit form

OwenLobato committed:
modify getOrganizationById with Client info

OwenLobato committed:
Add superadmin can edit an organization

OwenLobato committed:
Add edit client on firebase

OwenLobato committed:
Add edition of password and passphrase on Mongo and Firebase

EsquizoDEV commented on functions/components/organization/store.js:
```diff
@@ -78,7 +71,41 @@ export const getOrganizationById = (id, clean=false) => {
+            if (result.length === 0) {
+                reject("Organization not found");
+            } else {
+                resolve(result[0]);
```
Here we need to hydrate the resullting object if the function is not receiving the clean option, because the object returned is a plain javascript object when an aggregation its executed using Mongoose (and raw mongo queries), and there are other controllers that are using this function to query the organization which perform operations over it. For example is blocking the product creation because the method save() doesnt exist for plain objects.

This problem can be solved with the following line of code:

orgModel.hydrate(result[0]) and return this if the consumer of the function has not passed the clean parameter

EsquizoDEV commented on functions/components/security/network.js:
```diff
@@ -207,7 +206,6 @@ authRouter.post('/refresh', (req, res) => {
             } else {
                 getOrganizationById(claims.organization)
```
This kind of situations is where the getOrganizationById consumers must request the clean object without hydration, which includes properties and methods that corresponds to a Mongoose Object. Could we please correct this to work with the solution provided above?

OwenLobato committed:
Fix get organiaztion by id with hydrate

OwenLobato committed:
Add missing schema data to organization

OwenLobato committed:
Fix underscore on the produt name

OwenLobato committed:
Fix customers not required when organization is created

OwenLobato committed:
Add phone extension

OwenLobato committed:
Add container address can be the same as organization address

OwenLobato committed:
Fix superadmin firease phone

OwenLobato committed:
Add rollback on firebase if error

OwenLobato committed:
Delete user, org and pass if process fails

OwenLobato committed:
Add error messages

OwenLobato committed:
Add input validator to organization form (admin)

OwenLobato committed:
Fix NewOrganization form
- Fix passphrase regex
- Change checkbox postition and add tooltip
- Add customer validation

OwenLobato committed:
Add new employee input validator

OwenLobato committed:
Add clean prop to obtain org
